Flick/Controllers/TabBarController.swift

```diff
@@ -46,4 +47,68 @@ class TabBarController: UITabBarController {
         viewControllers = tabBarList
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if !UserDefaults.standard.bool(forKey: Constants.UserDefaults.didPromptPermission) {
+            showPermissionModal()
+        }
+    }
+
+}
+
+extension TabBarController: SPPermissionsDataSource, SPPermissionsDelegate {
+
+    func showPermissionModal() {
+        let controller = SPPermissions.dialog([.notification])
+        // Ovveride texts in controller
+        controller.titleText = "Permission Request"
+        controller.headerText = ""
+        controller.dataSource = self
+        controller.delegate = self
+        controller.present(on: self)
+    }
+
+    func configure(_ cell: SPPermissionTableViewCell, for permission: SPPermission) -> SPPermissionTableViewCell {
+        // Titles
+        cell.permissionTitleLabel.text = "Notifications"
+        cell.button.allowTitle = "Allow"
+        cell.button.allowedTitle = "Allowed"
+
+        // Colors
+        cell.iconView.color = .gradientPurple
+        cell.button.allowBackgroundColor = .gradientPurple
+        cell.button.allowedBackgroundColor = .gradientPurple
+        cell.button.allowTitleColor = .white
+        cell.button.allowedTitleColor = .white
+
+        return cell
+    }
+
+    func didAllow(permission: SPPermission) {
+        UserDefaults.standard.setValue(true, forKey: Constants.UserDefaults.didPromptPermission)
+        UIApplication.shared.registerForRemoteNotifications()
+    }
+
+    func didDenied(permission: SPPermission) {
+        UserDefaults.standard.setValue(true, forKey: Constants.UserDefaults.didPromptPermission)
```

Comment:
Can we just make sure to remove all of these from UserDefaults on logout? wonder if there is a way to just easily clear everything on logout haha